### PR TITLE
Admin - Support User Details  - update front end

### PR DESCRIPTION
**Pull request by rayking29**
# Implements [AB#15327](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15327)

## Admin - Support User Details  - update front end
- added new patient details page
- updated support page to make row links
- added HgField.razor to display detail fields (thanks Mike)
- Back button returns to support page and keeps search results
- any other route to support page clears search results

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

Screenshot 2023-05-04 at 2 04 43 PM

Screenshot 2023-05-04 at 2 03 58 PM

Screenshot 2023-05-04 at 2 09 56 PM

Screenshot 2023-05-04 at 2 10 49 PM

Screenshot 2023-05-04 at 2 11 22 PM

Screenshot 2023-05-04 at 2 53 33 PM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            <HgField Label="Name" Value="@PatientName" data-testid="patient-name" />
+        </MudItem>
+        <MudItem xs="12" lg="4">
+            <HgField Label="Birthdate" Value="@Patient.Birthdate.ToString()" data-testid="patient-dob" />
+        </MudItem>
+        <MudItem xs="12" lg="4">
+            <HgField Label="PHN" Value="@Patient.PersonalHealthNumber" data-testid="patient-phn" />
+        </MudItem>
+        <MudItem xs="12">
+            <HgField Label="HDID" Value="@Patient?.Hdid" data-testid="patient-hdid" />
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/PatientDetailsPage.razor:69]: Conditional access should not be needed inside this if block.

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+</HgBannerFeedback>
+
+<HgBannerFeedback
+    Severity="@Severity.Info"
+    IsEnabled="HasPatientsWarning"
+    TResetAction="PatientSupportActions.ResetStateAction"
+    Class="mt-3"
+    DataTestId="user-banner-feedback-warning-message">
+    <ul>
+        @foreach (string warning in PatientSupportState.Value.WarningMessages)
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/PatientDetailsPage.razor:39]: Nit: Could consider mapping this to a property called WarningMessages.

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    </ul>
+</HgBannerFeedback>
+
+<HgBannerFeedback
+    Severity="@Severity.Info"
+    IsEnabled="@(Warning != null)"
+    TResetAction="PatientSupportActions.ResetStateAction"
+    Class="mt-3"
+    DataTestId="user-banner-feedback-status-warning-message">
+    @Warning
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/PatientDetailsPage.razor:52]: Nit: Could consider renaming this to StatusWarning.

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            };
+        }
+
+        private void RetrievePatientData()
+        {
+            if (this.Patient?.Hdid != this.Hdid)
+            {
+                this.Dispatcher.Dispatch(new PatientSupportActions.ResetStateAction());
+                this.Dispatcher.Dispatch(new PatientSupportActions.LoadAction(PatientQueryType.Hdid, this.Hdid));
+            }
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs:106]: This if statement can be simplified to `this.Patient == null`.

**Comment by ninosamson** [comment]
If they are not a user, but in EMPI with HDID, should the HDID still show @rayking29 ?

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+            if (this.Patient?.Status == PatientStatus.Default)
+            {
+                this.Dispatcher.Dispatch(new MessageVerificationActions.ResetStateAction());
+                this.Dispatcher.Dispatch(new MessageVerificationActions.LoadAction(this.Hdid));
+            }
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs:112]: I think these should be moved into the previous if block and the status check should be removed.

1. Only patients with a status of NotUser are guaranteed to have no UserProfiles.
2. If MessageVerificationActions.ResetStateAction is not called, you may end up with leftover messaging verifications still being displayed when querying a different patient
3. Due to the asynchronous nature of the calls, Patient may be null when this executes.
   - If you wanted or needed to issue the calls sequentially, you could subscribe to PatientSupportActions.LoadSuccessAction during this page's initialization, and have it call a method that dispatches MessageVerificationActions.LoadAction.

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor (hunk 1 above)
Comment (rayking29) [reply Apps/Admin/Client/Pages/PatientDetailsPage.razor:69]: I'm not sure what you mean.

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor (hunk 3 above)
Comment (rayking29) [reply Apps/Admin/Client/Pages/PatientDetailsPage.razor]: Done

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs (hunk 4 above)
Comment (rayking29) [reply Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs]: Done

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs (hunk 5 above)
Comment (rayking29) [reply Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs]: Done

File: Apps/Admin/Client/Pages/PatientDetailsPage.razor [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            <HgField Label="Name" Value="@PatientName" data-testid="patient-name" />
+        </MudItem>
+        <MudItem xs="12" lg="4">
+            <HgField Label="Birthdate" Value="@Patient.Birthdate.ToString()" data-testid="patient-dob" />
+        </MudItem>
+        <MudItem xs="12" lg="4">
+            <HgField Label="PHN" Value="@Patient.PersonalHealthNumber" data-testid="patient-phn" />
+        </MudItem>
+        <MudItem xs="12">
+            <HgField Label="HDID" Value="@Patient?.Hdid" data-testid="patient-hdid" />
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/PatientDetailsPage.razor:69]:
```suggestion
            <HgField Label="HDID" Value="@Patient.Hdid" data-testid="patient-hdid" />
```

**Review by MikeLyttle — DISMISSED** [review]

**Review by MikeLyttle — APPROVED** [review]

**Review by ninosamson — COMMENTED** [review]
Looks good.

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**